content/en/agent/faq/linux-agent-2022-key-rotation.md

```diff
@@ -0,0 +1,137 @@
+
+Customers using Datadog's RPM or DEB packages might require a manual action to import the new key on their systems to install or upgrade the Agent after the rotation takes place.
+
+<div class="alert alert-info">
+<strong>Note</strong>: This DOES NOT affect the functionality of already running Agents, and only limits the ability to install or upgrade to a newer version of the Agent. Also, this doesn't affect Dockerized Linux Agents, Windows, or macOS Agents.
+</div>
+
+## Install methods that automatically trust the new GPG key
+
+Your host automatically trusts the new key (no further action is required) if you're using one of the following install methods:
+
+- [Agent install script][5] v1.6.0+ (released Jul 26, 2021)
+- [Chef cookbook][6] v4.11.0+ (released Aug 10, 2021)
+- [Ansible role][7] v4.10.0+ (released May 25, 2021)
+- [Puppet module][8] v3.13.0+ (released Aug 11, 2021)
+- [SaltStack formula][9] v3.4+ (released Aug 12, 2021)
+- [Heroku buildpack][10] v1.26+ (released May 26, 2021)
+- [Elastic Beanstalk][11] config templates updated as of Mar 29, 2021 or later (should contain `DATADOG_RPM_KEY_FD4BF915.public` under `gpgkey`)
+- Containerized Agents (Docker/Kubernetes) for any version
+- Windows/MacOS Agents for any version
+
+Additionally, if you are using the DEB package and Agent v7.31.0+, your hosts should have the `datadog-signing-keys` package installed, which automatically adds the new key (no further action is needed).
+
+If you're unsure if a host trusts the new signing key, you can [check](#check-if-a-host-trusts-the-new-gpg-key).
+
+For hosts running older versions of the install methods listed above or older versions of the DEB package, Datadog recommends updating the install method to the latest version. Alternatively Debian and Ubuntu users can update the Agent to version 7.31.0+. Otherwise, the key can be [manually updated](#manual-update).
+
+## What happens if I don't trust the new key before it is rotated?
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Trying to install or upgrade Agent packages without trusting the new key results in `NO_PUBKEY` errors. This applies to both newly released and existing versions of the Agent.
+
+```
+The following signatures couldn't be verified because the public key is not available: NO_PUBKEY
+```
+
+{{% /tab %}}
+{{% tab "RedHat/CentOS/SUSE" %}}
+
+Installing new versions of the Agent released since April 2022 causes `NOKEY` errors. Older versions of the Agent can still be installed.
+
+```
+The GPG keys listed for the "Datadog, Inc." repository are already installed but they are not correct for this package.
+Check that the correct key URLs are configured for this repository.
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Manual update
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Run the following commands on the host:
+
+```bash
+$ curl -o /tmp/DATADOG_APT_KEY_F14F620E https://keys.datadoghq.com/DATADOG_APT_KEY_F14F620E.public
+$ sudo apt-key add /tmp/DATADOG_APT_KEY_F14F620E
```

Comment:
We might add a comment saying that this is only necessary on Debian < 9 and Ubuntu < 16, but it's not strictly necessary.

Reply:
I think it's better to keep it simple and not add more conditionals.

Reply:
I agree with not adding conditionals. I just think users might wonder why the key has to be added twice, so a comment explaining it could be helpful.

Reply:
It's a bit hard to explain because it's a bit more confusing than "only necessary on Debian < 9 and Ubuntu < 16". 

Actually, the only part that's needed on all distros is the `apt-key add` because the keyring file is not going to be used by the `sources.list.d` file: if it is used it means it was created with an install method that also created the keyring, so this is not needed.

The only reason I'm adding the keyring file is because the way we are telling people to check if a system is in a good/bad state is to check for the file. The keyring file could even be empty and everything would still hold.

Reply:
Oh you're right, we would need to modify the sources.list.d file, which would be even more complicated, so it's probably best to use apt-key add 👍